app/controlplane/extensions/core/template/v1/extension.go

```diff
@@ -0,0 +1,197 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package template
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/extensions/core/template/v1/api"
+	"github.com/chainloop-dev/chainloop/app/controlplane/extensions/sdk/v1"
+	"github.com/go-kratos/kratos/v2/log"
+)
+
+type Integration struct {
+	*sdk.FanOutIntegration
+}
+
+// You can use an arbitrary struct as a configuration state
+// type registrationState struct{}
+// type attachmentState struct{}
+
+// Attach attaches the integration service to the given grpc server.
+// In the future this will be a plugin entrypoint
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanout(
+		&sdk.NewParams{
+			ID:      "template",
+			Version: "1.0",
+			Logger:  l,
+		},
+		// You can specify the inputs this attestation will be subscribed to, materials and or attestation envelope.
+		// In this case we are subscribing to SBOM_CYCLONEDX_JSON
+		sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON),
+		// You can also subscribed to any kind of materials
+		// sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_MATERIAL_TYPE_UNSPECIFIED),
```

Comment:
What if you want both all materials and attestation?

Reply:
You can put both. It's commented out.